docs: create 'Contribute to Documentation' guide

shivansh-gohem opened the pull request:
Hi @eeshaanSA and @khanhtc1202,

As discussed in issue #6124, this PR adds the new "Contribute to PipeCD Documentation" guide.

Following the project's documentation workflow, I've added the new file (`contributing-documentation.md`) to both:

* `docs/content/en/docs-dev/`
* `docs/content/en/docs-v1.0.x/`

This new page provides a clear guide for new contributors, covering:
* Where to find the documentation files
* How to build the site locally (`make run/site`)
* The standard PR process for doc changes

Here is a screenshot of the new page running locally:

<img width="1919" height="902" alt="Screenshot 2025-10-27 162727" src="https://github.com/user-attachments/assets/1ebc8dfa-05d4-4e25-8b22-244dc02f4cab" />

Please let me know if you'd like any changes!

eeshaanSA reviewed (changes requested):
Additionally, it would be also great to explicitly mention that our documentation resides on https://pipecd.dev, our website.

eeshaanSA commented:
@shivansh-gohem Sorry for the delay in response. 

The overall structure looks good, just some minor changes and this should be merged into the main. Thank you for your contribution. 👍

shivansh-gohem commented:
Thanks, @eeshaanSA! I've updated the metadata in both files to include description and the correct weight: 4.

I've also rebased the branch to resolve the 'out-of-date' warning. Please let me know if this looks good!

eeshaanSA commented on docs/content/en/docs-dev/contribution-guidelines/contributing-documentation.md:
```diff
@@ -0,0 +1,40 @@
+---
```
```suggestion
---
title: "Contribute to PipeCD Documentation"
linkTitle: "Contribute to PipeCD Documentation"
description: > 
  A guide for how to contribute to the PipeCD Documentation website (https://pipecd.dev).
```

eeshaanSA replied:
@shivansh-gohem this is the metadata that we follow. This was present in the file I linked as well. 👀 

Please follow this from the next time. 😄

eeshaanSA replied:
Also, all hyperlinks are better off starting with https://.

eeshaanSA commented on docs/content/en/docs-dev/contribution-guidelines/contributing-documentation.md:
```diff
@@ -0,0 +1,40 @@
+title: "Contribute to PipeCD Documentation"
+linkTitle: "Contribute to PipeCD Documentation"
+description: >
+  A guide for how to contribute to the PipeCD Documentation website (https://pipecd.dev).
```
```suggestion
This page describes how to contribute to the PipeCD Documentation.
```

I think the description is better off this way. The current drscp looks cluttered on the website. 

<img width="562" height="77" alt="Image" src="https://github.com/user-attachments/assets/6f405fa3-60d4-471f-bdb3-975d88c07995" />

eeshaanSA replied:
fix this nit, and we are good to go, @shivansh-gohem

shivansh-gohem commented:
All changes are done!!

eeshaanSA commented:
LGTM, Thanks @shivansh-gohem

khanhtc1202 reviewed (approved):
Thanks @shivansh-gohem 🙌
